spec.bs

```diff
@@ -0,0 +1,369 @@
+
+  Note: The <a href="https://github.com/patcg-individual-drafts/topics/blob/main/taxonomy_v1.md">initial taxonomy</a> for Chrome experimentation includes hundreds of topics. The taxonomy to be used long term is TBD (<a href="https://github.com/patcg-individual-drafts/topics/issues/3">github issue</a>). 
+
+  The <dfn>model version</dfn> identifies the model used to classify a page into a [=topic=]. The meaning may vary across browser vendors.
+
+  The <dfn>configuration version</dfn> identifies the algorithm (other than the model part) used to calculate the topic. The meaning may vary across browser vendors.
+
+  The <dfn>version</dfn> is the concatenation of browser vendor name, [=configuration version=], [=taxonomy version=], and [=model version=].
+
+  <div class="example">
+  In Chrome, the [=version=] takes the form of <code>chrome.[=configuration version=]:[=taxonomy version=]:[=model version=]</code>.
+  <div>
+
+</section>
+
+<section>
+  <h2 id="topics-caller-context-header">Topics caller context</h2>
+  A <dfn>topics-caller-context</dfn> comprises the following fields:
+  1. <dfn>caller-origin</dfn>
+      - For document.browsingTopics() request, the [=caller-origin=] is the document's origin.
+      - For navigation or fetch() request, the [=caller-origin=] is the ongoing request URL's origin (and therefore can change across redirects).
+  1. <dfn>page</dfn>
+      - For document.browsingTopics() request, the [=page=] is the top level page associagted with the document.
+      - For fetch() reqeust initiated from a Window context, the [=page=] is the top level page associated with the fetch() initiator Window.
+      - For navigation request:
+          - If the navigation happens on an iframe:
+            - The [=page=] is the top level page associated with the iframe.
+          - Else (i.e. the navigation happens on the top level browsing context):
+            - The [=page=] is the page that the navigation would result in. Its top level URL should equal the navigation URL.
+
+  Note: a [=page=] can be inactive (e.g. pre-rendered; intermediate redirect for a top navigation), but under all cases it should have a top level URL associated.
+
+  <h2 id="determine-the-topics-of-a-page-header">Determine the topics of a page</h2>
+  The browser should use a model to classify a [=page=] into [=topics=] that are most relevant to the page. The <dfn>input data</dfn> for the model may be the page's domain, URL, text content, etc.
+
+  <h2 id="collect-page-topics-calculation-input-data-header">Collect page topics calculation input data</h2>
+  Given a [=page=] |P|, the <dfn>collect page topics calculation input data</dfn> method steps are:
+  1. Run the following steps in parallel:
+    1. Wait until the necessary [=input data=] |D| for the |P| is ready.
+    1. Wait until |P| has committed and is active.
```

Comment:
What does this mean? It needs a rigorous definition or reference to an existing rigorous definition.